### DISCUSSION TITLE
請問keybinder中的when條件可以自定義嗎?

### DISCUSSION THREAD

**Discussion by cs10108** [rime/home#1539] [General]
目前keybinding的when只有4種選項 我想做成這樣
when: "custom rule", accept: XXXXX, send: XXX

起因是這樣的 我想用rime取代巨硬的速成輸入法
找到一個適合我用的傳統速成
https://github.com/philipposkhos/rime-ms-quick
但作者也知道問題
"不足 : 當直接輸入單字碼字時(例如 手, 火等)，無法像 Windows 輸入法那樣直接用 space 輸入，需要像選字那樣按 1 字。"
具體分別是這樣的
巨硬:
單按a-z 不出侯選詞
單按a-z加空格而無重碼上屏碼字 例如:Q空格 = 手
單按a-z加空格如有重碼出侯選詞 例如:A空格 = 1. 日 / 2. 曰
Rime:
單按a-z會出侯選字 而碼字則是侯選詞的第1格
單按a-z無法用空格上屏碼字 因為when has_menu 空格被綁定為page down

想改善此問題 我有個折衷方案
預設space已經能把單碼上屏 那要麼我改雙碼的設定
在keybinding中自定義when的條件
例如when 已輸入雙碼出現侯選詞時 accept space send page down

否則只能考慮大改輸入法 完全還原巨硬

**Comment by ksqsf** [comment]
按我的理解，用 Rime 自帶的組件不太容易實現這個效果， `key_binder` 的條件沒那麼靈活。

如果一定需要類似的效果，可以用 C++ 實現編寫插件實現相應邏輯，當然也可以使用 [Lua 插件](https://github.com/hchunhui/librime-lua/wiki/Scripting)。
